src/mlpack/tests/recurrent_network_test.cpp

```diff
@@ -1413,4 +1413,68 @@ BOOST_AUTO_TEST_CASE(BRNNTrainReturnObjective)
   BOOST_REQUIRE_EQUAL(std::isfinite(objVal), true);
 }
 
+/**
+ * Test that RNN::Train() does not give an error for large rho.
+ */
+BOOST_AUTO_TEST_CASE(LargeRhoValueRnnTest)
+{
+  const int rho = 18;
+  const int hiddenSize = 128;
+  const int numLetters = 256;
+  using namespace arma;
+  using MatType = cube;
+  std::vector<std::string> trainingData;
+  trainingData.push_back(std::string("THIS IS THE INPUT"));
+
+  RNN<> model(rho);
+  model.Add<IdentityLayer<>>();
+  model.Add<LSTM<>>(numLetters, hiddenSize, rho);
+  model.Add<Dropout<>>(0.1);
+  model.Add<Linear<>>(hiddenSize, numLetters);
+
+  const auto makeInput = [](const char *line) -> MatType {
+    const auto strLen = strlen(line);
+    // rows: number of dimensions
+    // cols: number of sequences/points
+    // slices: number of steps in sequences
+    MatType result(numLetters, 1, strLen, fill::zeros);
+    for (int i = 0; i < strLen; ++i)
+    {
+      const auto letter = line[i];
+      result.at(static_cast<uword>(letter), 0, i) = 1.0;
+    }
+    return result;
+  };
+
+  const auto makeTarget = [] (const char *line) -> MatType {
+    const auto strLen = strlen(line);
+    // responses for NegativeLogLikelihood should be
+    // non-one-hot-encoded class IDs (from 1 to num_classes)
```

Comment:
This is a little pedantic (sorry about that) but the mlpack style guide suggests that comments be complete sentences (with punctuation and everything) so that people for whom English is not a native language can have an easier time with it.  So, e.g., `Responses for NegativeLogLikelihood should be non-one-hot-encoded class IDs (from 1 to num_classes).`  Are you willing to change the comments here to match that?  Alternately, I can do it during merge.